src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -562,11 +562,13 @@ else if (ast.getType() == TokenTypes.SWITCH_RULE) {
     private static DetailAST findLastChildWhichContainsSpecifiedToken(DetailAST ast, int childType,
                                                               int containType) {
         DetailAST returnValue = null;
-        for (DetailAST astIterator = ast.getFirstChild(); astIterator != null;
-                astIterator = astIterator.getNextSibling()) {
+        for (DetailAST astIterator = ast.getLastChild();
+                ;
```

Comment:
http://rveach.no-ip.org/checkstyle/regression/302/
No differences in regression. If this was an issue, we would have gotten a NPE.

Reply:
TC will not allow code like this.
> 565: findLastChildWhichContainsSpecifiedToken() for statement lacks condition

Reply:
Without removing the condition we get missing code coverage.

@romani How should I handle this?

Reply:
Code is weird without condition, can you share coverage problems?

Reply:
Below is the code coverage with the code changes.
![Image1](https://user-images.githubusercontent.com/5427943/148121504-2035edb4-87af-4716-a8c4-729250240907.png)

So everything is covered except the end condition, meaning we are always ending the loop before we reach the point where `astIterator` can equal `null`. `astIterator` can only be `null` when `getPreviousSibling` returns `null` meaning we have searched through all the children and have not found the requested `childType` and `containType`.

To me, this seems to indicate we are already being too strict before this method is called on what we call it on so maybe `childType` or `containType`.

It could be `childType` as I am already starting to look at pitest and expanding it to a new mutator in another issue.
Pitest report is the following, but it could be changed as I actively work on the issue.
http://rveach.no-ip.org/checkstyle/pitest/40/pitest-coding/com.puppycrawl.tools.checkstyle.checks.coding/FinalLocalVariableCheck.java.html#org.pitest.mutationtest.report.html.SourceFile@5ac60f01_566

Reply:
I would keep permanent suppression on this spotbugs rules.

Reply:
Or keep it, but suppress this violation. So they will be two suppression.

Reply:
Done.